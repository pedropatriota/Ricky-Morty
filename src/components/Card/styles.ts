import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	flex-direction: column;
	width: 18rem;
`;

export const ImagContainer = styled.div<{ image: string }>`
	background-image: url(${props => props.image});
	background-repeat: no-repeat;
	background-size: cover;
	width: 100%;
	max-width: 18rem;
	height: 18rem;
	border-radius: 1rem 1rem 0 0;
`;

export const InfoContainer = styled.div`
	padding: 0.5rem;
	background-color: ${props => props.theme.colors.el};
	border: 1px solid ${props => props.theme.colors.el};
	border-radius: 0 0 1rem 1rem;

	h2 {
		margin-bottom: 1rem;
		color: ${props => props.theme.colors.text};
	}
	p {
		margin-bottom: 0.4rem;
		color: ${props => props.theme.colors.text};
	}
`;
