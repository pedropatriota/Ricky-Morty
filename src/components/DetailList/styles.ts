import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	gap: 0.75rem;
	flex-direction: column;
	margin-bottom: 0.75rem;
`;
