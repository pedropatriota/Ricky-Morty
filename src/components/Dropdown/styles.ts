import styled from 'styled-components';
import Select from 'react-select';

export const ReactSelect = styled(Select)`
  .react-select__control {
    width: min(90vw, 12.5rem);
    background-color: ${(props) => props.theme.colors.el};
    border-color: ${(props) => props.theme.colors.border};
    box-shadow: 0 0 0 transparent;
    height: 3.375rem;

    &:hover {
      border-color: ${(props) => props.theme.colors.border};
    }
  }

  .react-select__single-value {
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__menu {
    width: 12.5rem;
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__option,
  .react-select__option--is-focused 
  {
    background-color: ${(props) => props.theme.colors.el};
    color: ${(props) => props.theme.colors.text};
  }

  .react-select__dropdown-indicator,
  .react-select__indicator-separator {
    display: none;
  }
`;
