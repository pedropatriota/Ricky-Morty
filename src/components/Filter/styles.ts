import styled from 'styled-components';

export const Container = styled.div`
	display: flex;
	align-items: flex-end;
	justify-content: space-between;
	flex-wrap: wrap;
	gap: 0.8rem;

	@media (max-width: 500px) {
		flex-wrap: wrap;
		gap: 1rem;
	}
`;
