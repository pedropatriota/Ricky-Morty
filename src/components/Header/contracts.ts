export type TTheme = {
	title: string;
	colors: {
		bg: string;
		el: string;
		text: string;
		input: string;
		border: string;
	};
};

export interface HeaderProps {
	theme: TTheme;
	toggleTheme: () => void;
}
