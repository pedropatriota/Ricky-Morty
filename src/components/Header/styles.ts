import styled from 'styled-components';

export const Header = styled.header`
	padding: 0px min(5vw, 5rem);
	background-color: ${({ theme }) => theme.colors.bg};
	box-shadow: 0px 10px 15px -3px rgba(0, 0, 0, 0.1);
	margin-bottom: 4rem;
`;

export const Container = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	flex-wrap: wrap;

	img {
		width: 100%;
		height: auto;
		max-width: 10rem;
	}
`;
