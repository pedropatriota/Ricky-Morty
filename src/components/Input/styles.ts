import styled from 'styled-components';

export const Input = styled.input`
  width: min(90vw, 30rem);
  border: ${(props) => props.theme.colors.border};
  color: ${(props) => props.theme.colors.text};
  height: 3.375rem;
  border-radius: 3px;
  padding: 0 2rem;
  font-size:0.94rem;
  outline: none;
  background: ${(props) => props.theme.colors.el}
`;
