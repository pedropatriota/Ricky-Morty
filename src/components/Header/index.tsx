import * as Styled from './styles';
import rickyMortyLogo from './RickAndMortyLogo.png';
import { Moon, MoonStar } from 'lucide-react';
import { HeaderProps } from './contracts';

const Header = ({ theme, toggleTheme }: HeaderProps) => {
	return (
		<Styled.Header>
			<Styled.Container>
				<img src={rickyMortyLogo} alt="Ricky and Morty logo" width="160px" />
				<div onClick={toggleTheme} role="button" tabIndex={0}>
					{theme.title === 'dark' ? (
						<MoonStar data-testid="icon-dark" />
					) : (
						<Moon data-testid="icon-light" />
					)}
				</div>
			</Styled.Container>
		</Styled.Header>
	);
};

export default Header;
