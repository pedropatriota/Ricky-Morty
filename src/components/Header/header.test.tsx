import { screen, render, within, fireEvent, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { light, dark } from '../../styles/themes';
import Header from '.';

const setup = () => {
	const toggleTheme = jest.fn();
	const utils = render(
		<ThemeProvider theme={light}>
			<Header toggleTheme={toggleTheme} theme={light} />
		</ThemeProvider>
	);
	const header = screen.getByRole('banner');

	return {
		header,
		toggleTheme,
		...utils
	};
};

afterEach(cleanup);

describe('Header', () => {
	const toggleThemeMocked = jest.fn();
	beforeEach(() => {
		toggleThemeMocked.mockClear();
	});

	it('should render Header component', () => {
		const { header } = setup();

		const logo = within(header).getByRole('img', {
			name: /ricky and morty logo/i
		});
		const toggleThemeBtn = within(header).getByRole('button');
		const lightSvg = screen.getByTestId('icon-light');

		expect(logo).toBeInTheDocument();
		expect(toggleThemeBtn).toBeInTheDocument();
		expect(lightSvg).toHaveClass('lucide lucide-moon');
		expect(header).toHaveStyle(`background-color: ${light.colors.bg}`);
	});

	it('should change the theme when toggleThemeBtn is clicked', async () => {
		const { header, toggleTheme } = setup();

		const lightSvg = screen.queryByTestId('icon-light');
		const toggleThemeBtn = within(header).getByRole('button');

		expect(lightSvg).toBeInTheDocument();

		fireEvent.click(toggleThemeBtn);

		expect(toggleTheme).toHaveBeenCalled();
	});

	it('should match the snapshot', () => {
		const { container } = setup();
		expect(container).toMatchSnapshot();
	});
});
