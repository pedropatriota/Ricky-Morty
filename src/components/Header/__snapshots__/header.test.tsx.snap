// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<div>
  <header
    class="sc-aXZVg bwHViG"
  >
    <div
      class="sc-gEvEer hPDMNs"
    >
      <img
        alt="Ricky and Morty logo"
        src="RickAndMortyLogo.png"
        width="160px"
      />
      <div
        role="button"
        tabindex="0"
      >
        <svg
          class="lucide lucide-moon"
          data-testid="icon-light"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
          />
        </svg>
      </div>
    </div>
  </header>
</div>
`;
